k256/src/arithmetic/field.rs

```diff
@@ -542,6 +544,28 @@ mod tests {
         assert_eq!(four.sqrt().unwrap().normalize(), two.normalize());
     }
 
+    #[test]
+    fn unnormalized_is_odd() {
+        // This is a regression test for https://github.com/RustCrypto/elliptic-curves/issues/529
+        // where `is_odd()` in debug mode force-normalized its argument
+        // instead of checking that it is already normalized.
+        // As a result, in release (where normalization didn't happen) `is_odd()`
+        // could return an incorrect value.
+
+        let x = FieldElement::from_bytes_unchecked(&[
+            61, 128, 156, 189, 241, 12, 174, 4, 80, 52, 238, 78, 188, 251, 9, 188, 95, 115, 38, 6,
+            212, 168, 175, 174, 211, 232, 208, 14, 182, 45, 59, 122,
+        ]);
+        // Produces an unnormalized FieldElement with magnitude 1
+        // (we cannot create one directly).
+        let y = x.sqrt().unwrap();
+        // Panics since `y` is not normalized.
+        let result = std::panic::catch_unwind(|| y.is_odd());
+        assert!(result.is_err());
```

Comment:
`std` isn't always going to be available in unit tests, since this is a `no_std` crate.

There are a couple options here to keep the tests `no_std`-friendly:

1. Write a `#[should_panic]` unit test: https://doc.rust-lang.org/reference/attributes/testing.html#the-should_panic-attribute
2. Use an integration test in `tests/` where `std` is linked unconditionally

I would personally suggest the first option

Reply:
Yeah, I just noticed that fails too. On it.

Reply:
IIRC it's not trivial to run tests in a `no_std` environment, since the `test` crates currently depends on `std`.

A bigger issue is that tests may be compiled with `panic="abort"`, with which `catch_unwind` obviously will not work.

I agree with going with the first option.

Reply:
I guess those fails are with `panic="abort"` where the test doesn't panic. Any ideas how to gate it?